src/Util/ClassNameValue.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+use Symfony\Bundle\MakerBundle\Str;
+
+/**
+ * @internal
+ */
+final class ClassNameValue
+{
+    private $fullClassName;
+    private $typeHint;
+
+    public function __construct(string $fullClassName, string $typeHint)
+    {
+        $this->fullClassName = $fullClassName;
+        $this->typeHint = $typeHint;
+    }
+
+    public function getShortName(): string
+    {
+        if ('self' === $this->typeHint) {
+            return Str::getShortClassName($this->fullClassName);
+        }
+
+        return $this->typeHint;
+    }
+
+    public function __toString()
```

Comment:
Can we avoid having a `__toString()`? I'd prefer to be explicit and call `getShortName()`